.github/workflows/label-mandatory-workspace-prs.yaml

```diff
@@ -0,0 +1,243 @@
+                    case 'ADDITION':
+                      description = 'PR adds a new workspace';
+                      color = '0e8a16';  // Green
+                      break;
+                    case 'OUTSIDE':
+                      description = 'PR changes files outside workspace directories';
+                      color = '6f42c1';  // Purple
+                      break;
+                    case 'MANDATORY':
+                      description = 'PR affects a workspace with required plugins for releases';
+                      color = 'd73a4a';  // Red
+                      break;
+                  }
+                  
+                  await github.rest.issues.createLabel({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    name: labelName,
+                    description: description,
+                    color: color
+                  });
+                  console.log(`Created label: ${labelName}`);
+                } else {
+                  throw error;
+                }
+              }
+            }
+            
+            // Get all open PRs
+            const prs = await github.paginate(github.rest.pulls.list, {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              state: 'open',
+              per_page: 100
+            });
+            
+            console.log(`Found ${prs.length} open PRs`);
+            
+            for (const pr of prs) {
+              try {
+                console.log(`\n--- Processing PR #${pr.number}: ${pr.title} ---`);
+                
+                // Get current labels on the PR
+                const currentLabels = pr.labels.map(label => label.name);
+                const currentWorkspaceLabels = currentLabels.filter(label => 
+                  Object.values(LABELS).includes(label)
+                );
+                
+                // Analyze PR files to know what changes this PR contains
+                const prFiles = await github.rest.pulls.listFiles({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  pull_number: pr.number
+                });
+                
+                // Categorize files
+                const workspaceFiles = [];
+                const nonWorkspaceFiles = [];
+                const addedWorkspaces = new Set();
+                const modifiedWorkspaces = new Set();
+                
+                for (const file of prFiles.data) {
+                  const workspaceMatch = file.filename.match(/^workspaces\/([^\/]+)\/.*/);
+                  if (workspaceMatch) {
+                    const workspace = workspaceMatch[1];
+                    workspaceFiles.push({ file, workspace });
+                    
+                    // Check if this is a new workspace
+                    if (file.status === 'added') {
+                      addedWorkspaces.add(workspace);
```

Comment:
Is this test sufficient to know that it is the addition of a full workspace that didn't exist before ?
What happens if the PR only add a new file to an already-existing workspace (like a patch file in the `/patches` sub-folder) ?